### PR TITLE
Create Qemu command wrapper

### PR DESCRIPTION
**Pull request by jakecorrenti**
Creates a wrapper around the Qemu command line implementation to prevent the need to hard-code the different command line options in Init and Start.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/machine/qemu/command.go
Comment (baude) [inline pkg/machine/qemu/command.go:98]: should this return []string ?

Comment (jakecorrenti) [reply pkg/machine/qemu/command.go:98]: Yes, I believe you're right. I will fix that

**Review by ashley-cui — COMMENTED** [review]
LGTM

File: capture.pcap
Comment (flouthoc) [inline capture.pcap:1]: Is this file needed ? Why is `capture.pcap` needed in the repo ?

**Review by flouthoc — CHANGES_REQUESTED** [review]
One comment above I think that needs to be addressed otherwise LGTM

File: pkg/machine/e2e/qemu_command_test.go
Comment (Luap99) [inline pkg/machine/e2e/qemu_command_test.go:25]: This is not a e2e test, this is an unit test.
As such this should be moved into the qemu dir and turned into a regual unit test, first these tests here run on real hardware so time is much more expensive and the setup here is just wasted time.

File: pkg/machine/qemu/command.go
Comment (Luap99) [inline pkg/machine/qemu/command.go:95]: this is not used so please remove it.

File: pkg/machine/qemu/command.go
Comment (Luap99) [inline pkg/machine/qemu/command.go:25]: if you accept unit you should not cast back to int, this will just cause troubles when you overflow the type.
use `FormatUint(m, 10)`

I know pre-existing but still.

File: pkg/machine/qemu/command.go
Comment (Luap99) [inline pkg/machine/qemu/command.go:31]: same here although I guess overflowing cpu's is unlikely

File: pkg/machine/qemu/machine.go
Comment (Luap99) [inline pkg/machine/qemu/machine.go:218]: if this uses the builder pattern than I don't think you need to assign the slice every time again,
just writing `v.CmdLine.WithVirtfsMount(source, tag, securityModel, readonly)` should already modify the slice in place

File: pkg/machine/qemu/machine.go
Comment (Luap99) [inline pkg/machine/qemu/machine.go:300]: same here

File: pkg/machine/qemu/machine.go
Comment (Luap99) [inline pkg/machine/qemu/machine.go:702]: and here

File: pkg/machine/qemu/machine.go
Comment (Luap99) [inline pkg/machine/qemu/machine.go:707]: and here

**Review by Luap99 — CHANGES_REQUESTED** [review]

File: pkg/machine/qemu/machine.go
Comment (baude) [reply pkg/machine/qemu/machine.go:218]: correct.

File: capture.pcap
Comment (TomSweeneyRedHat) [reply capture.pcap:1]: I'm not seeing it in the PR below, nor in the repo.  Generally, pcap files are used to capture network packets zinging by, and I don't see the need for it here.  Perhaps a file was left behind by Wireshark and accidentally committed?

**Comment by TomSweeneyRedHat** [comment]
I've added the do not merge/WIP label for the pcap file.  Once that file is removed or very fully explained and the other comments from others are addressed, feel free to remove the label.

File: capture.pcap
Comment (jakecorrenti) [reply capture.pcap:1]: the .pcap file is output when you do a command like `podman machine start --log-level=debug`. I left that in there accidentally, so I just have to remove it from the commit.

Comment (jakecorrenti) [reply capture.pcap:1]: I've just been on campus all day so I haven't had a chance to fix the PR yet

**Comment by jakecorrenti** [comment]
@Luap99 Thank you for the thorough review!

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**
